### ISSUE TITLE
 Instantiating separate instances of superagent-cache causes all requests to hang

### ISSUE THREAD

**Issue by kaspermarstal** [jpodwys/superagent-cache#7]
Hi,

Kudos for putting this little gem of a library together.

We want to use it for caching requests to Google Books API but have encountered a small issue during development. When two separate instances of superagent-cache is instantiated the application hangs indefinitely. 

We have put together a little example to reproduce the problem at https://github.com/ilovebike/googlebooks/tree/development/. Clone and run "npm test" the tests complete just fine. However, uncomment https://github.com/ilovebike/googlebooks/blob/development/test/search-redis.js#L8 and run the tests again, and notice that it hangs indefinitely when the first http request is dispatched. Even mochas default timeout functionality cannot break the lockup.

Is anyone able to reproduce the problem?

**Comment by jpodwys** [comment]
Thanks for trying the library out and thank you for the detailed bug description. I'm unable to attempt to reproduce this at the moment because I'm away from my computer, but I do have a question--why do you need more than one instance of superagent-cache running on a single machine?

**Comment by kaspermarstal** [comment]
Sounds great. Looking forward to hear about your findings. Have nice weekend!

**Comment by jpodwys** [comment]
Likewise!

**Comment by jpodwys** [comment]
I've created the `multiple-instances` branch to attempt to address this issue. I tried testing it in your repo, but there are some things it seems I don't have installed on my machine (such as babel) that are required to run your repo. Please try pinning temporarily to my branch and let me know if it fixes the issue. Thanks!

**Comment by kaspermarstal** [comment]
It works! Awesome! :grin: Thank you for your effort!

**Comment by jpodwys** [comment]
Published to NPM in `1.0.4`. Thanks for the bug report!

**Comment by kaspermarstal** [comment]
Hi again,

I may have jumped the gun on this. During the tests (in the repo linked to above, added the missing packages btw, hope it works now), the second instantiation with a redis-backed cache use the cache of the first instantiation. If the first instantiation is commented out or removed etc the redis cache works as expected.

**Comment by jpodwys** [comment]
OK, I made a new branch to attempt to fix the issue. Try using the `google-books-development` branch and let me know how it goes. I've made some notes about attempting to run your repo on my machine below.

When I try to run your repo locally, I get the following error:

```
docker-compose --file ./test/docker-compose.yml up -d
Couldn't connect to Docker daemon at http+docker://localunixsocket - is it running?
```

Additionally, this line doesn't look like it will play nicely on other people's machines due to the `redisUrl`:

```
superagent: superagentCache(null, redisCache({ redisUrl: 'http://user:pass@192.168.59.103:6379/'}))
```

`cache-service-redis` is capable of consuming a redis mock for testing. If you'd like, you can check out how I'm doing it [in `cache-service-redis`'s tests](https://github.com/jpodwys/cache-service-redis/blob/master/test/server/redis-cache-module.js#L2-L7). I should probably document this feature.

**Comment by kaspermarstal** [comment]
Great, tests now uses a redis mock as you suggested and with the new fixes the issue is now completely resolved. Thanks a lot!! We are really happy with the current setup.

**Comment by jpodwys** [comment]
Glad it's working now! Published to NPM in `1.0.5`.
